From 6dbd2ef938ea9de84b151d93b6addac510ba4d58 Mon Sep 17 00:00:00 2001
From: Jiangyi <sam.andrew.jiang@gmail.com>
Date: Sun, 13 Sep 2015 14:28:06 +0200
Subject: [PATCH] vendor/cm: Add chromium_prebuilt.sh

This script will pull the necessary libs from a complete build and generate proper
makefiles in order to support using a prebuilt Chromium. Files are placed in
prebuilts/chromium/-devicenamehere-, and are not tracked by git, so they are completely
out of the way.

PS3:
-replace git -C with params compatible to old git versions

Change-Id: Ib95434b87e1bcd288d645d3caba87d78ea3ee439
---
 config/common.mk           |  5 +++
 tools/chromium_prebuilt.sh | 81 ++++++++++++++++++++++++++++++++++++++++++++++
 2 files changed, 86 insertions(+)
 create mode 100755 tools/chromium_prebuilt.sh

diff --git a/config/common.mk b/config/common.mk
index f4a5de5..9a230b6 100644
--- a/config/common.mk
+++ b/config/common.mk
@@ -359,4 +359,9 @@ PRODUCT_PROPERTY_OVERRIDES += \
 
 -include vendor/cyngn/product.mk
 
+# Chromium Prebuilt
+ifeq ($(PRODUCT_PREBUILT_WEBVIEWCHROMIUM),yes)
+-include prebuilts/chromium/$(TARGET_DEVICE)/chromium_prebuilt.mk
+endif
+
 $(call prepend-product-if-exists, vendor/extra/product.mk)
diff --git a/tools/chromium_prebuilt.sh b/tools/chromium_prebuilt.sh
new file mode 100755
index 0000000..7a95d28
--- /dev/null
+++ b/tools/chromium_prebuilt.sh
@@ -0,0 +1,81 @@
+#!/bin/sh
+
+# Copyright (C) 2014 The OmniROM Project
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# This works, but there has to be a better way of reliably getting the root build directory...
+if [ $# -eq 1 ]; then
+    TOP=$1
+    DEVICE=$TARGET_DEVICE
+elif [ -n "$(gettop)" ]; then
+    TOP=$(gettop)
+    DEVICE=$(get_build_var TARGET_DEVICE)
+else
+    echo "Please run envsetup.sh and lunch before running this script,"
+    echo "or provide the build root directory as the first parameter."
+    return 1
+fi
+
+TARGET_DIR=$OUT
+PREBUILT_DIR=$TOP/prebuilts/chromium/$DEVICE
+
+if [ -d $PREBUILT_DIR ]; then
+    rm -rf $PREBUILT_DIR
+fi
+
+mkdir -p $PREBUILT_DIR
+mkdir -p $PREBUILT_DIR/app
+mkdir -p $PREBUILT_DIR/lib
+
+if [ -d $TARGET_DIR ]; then
+    echo "Copying files..."
+    cp -r $TARGET_DIR/system/app/webview $PREBUILT_DIR/app
+    cp $TARGET_DIR/system/lib/libwebviewchromium.so $PREBUILT_DIR/lib/libwebviewchromium.so
+    cp $TARGET_DIR/system/lib/libwebviewchromium_plat_support.so $PREBUILT_DIR/lib/libwebviewchromium_plat_support.so
+    cp $TARGET_DIR/system/lib/libwebviewchromium_loader.so $PREBUILT_DIR/lib/libwebviewchromium_loader.so
+else
+    echo "Please ensure that you have ran a full build prior to running this script!"
+    return 1;
+fi
+
+echo "Generating Makefiles..."
+
+HASH=$(git --git-dir=$TOP/external/chromium_org/.git --work-tree=$TOP/external/chromium_org rev-parse --verify HEAD)
+echo $HASH > $PREBUILT_DIR/hash.txt
+
+(cat << EOF) | sed s/__DEVICE__/$DEVICE/g > $PREBUILT_DIR/chromium_prebuilt.mk
+# Copyright (C) 2014 The OmniROM Project
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+LOCAL_PATH := prebuilts/chromium/__DEVICE__/
+
+\$(shell mkdir -p out/target/product/__DEVICE__/system/app/webview/lib/arm)
+\$(shell mkdir -p out/target/product/__DEVICE__/system/lib)
+\$(shell cp -r \$(LOCAL_PATH)/app/ out/target/product/__DEVICE__/system)
+\$(shell cp -r \$(LOCAL_PATH)/lib/ out/target/product/__DEVICE__/system)
+
+EOF
+
+echo "Done!"
-- 
2.5.1
